pandas/core/indexes/datetimes.py

```diff
@@ -199,10 +198,11 @@ def _join_i8_wrapper(joinf, **kwargs):
 
     _engine_type = libindex.DatetimeEngine
 
-    tz = None
+    _tz = None
```

Comment:
A `tz` property is defined below

Reply:
I am fairly certain that the version currently here is a mistake, since it is overriden by the property definition of `tz` below.  _freq and _tz are set in _simple_new, so I think the idea of also defining them on the class is to make introspecting what attributes exist easy.